doc/reference/vm-live-migration-internals.md

```diff
@@ -0,0 +1,112 @@
+5. Storage Driver Transfer.
+6. Start Target QEMU: Launch the QEMU process on the target.
+7. Add NBD Node: Attach the NBD target disk as a block node on the source. 
+8. Block Device Mirror: Perform a block device mirror of the snapshot to the target NBD server.
+9. Pause Source VM: Pause the source VM once the block device mirror enters `ready` state.
+10. State Transfer: Transfer the memory state over WebSocket.
+11. Enter `pre-switchover` State.
+12. Complete Migration: Continue state migration until `completed` state reached.
+
+```{figure} /images/vm_live_migration_flowchart.svg
+Non-Shared Storage Migration
+```
+
+The state transitions during the process are shown below:
+
+```{figure} /images/vm_live_migration_fsa_diagram.svg
+Non-Shared Storage Migration State Transitions
+```
+
+### Intra-Cluster Member Live Migration (Ceph Shared Storage Pool)
+
+For shared storage pools such as Ceph, disk data transfer is unnecessary. Instead, the process focuses on transferring the VM state through a dedicated migration socket.
+
+1. Validate cluster state and storage pool readiness.
+2. Transfer the live state data over the migration socket.
+3. Start the target VM using the transferred state.
+
+## Implementation Details
+
+### Migration API
+
+Sending a `POST` request to `/1.0/instances/{name}` renames, moves an instance between pools or migrates an instance to another server. The returned operation metadata for migration, in the push case, will be a background operation with progress data, for the pull case, it will be a WebSocket operation with a number of secrets to be passed to the target server.
+
+### Live Migration Call Stack
+
+Below is a general overview of the key functions part of the live migration call stack:
+
+#### [`lxd/lxd/instance_post.go`](https://github.com/canonical/lxd/lxd/instance_post.go)
+
+1. [`instancePost`](https://github.com/canonical/lxd/lxd/instance_post.go#L74)
+
+This function handles post requests to the `/1.0/instances` endpoint.
+
+#### [`lxd/lxd/migrate_instance.go`](https://github.com/canonical/lxd/lxd/migrate_instance.go)
+
+1. [`Do`](https://github.com/canonical/lxd/lxd/migrate_instance.go#L87)
+
+This function performs the migration operation on the source VM for the given state and operation. It sets up the necessary WebSocket connections for control, state, and filesystem, and then initiates the migration process.
+
+#### [`lxd/lxd/instance/drivers/driver_qemu.go`](https://github.com/canonical/lxd/lxd/instance/drivers/driver_qemu.go)
+
+1. [`MigrateSend`](https://github.com/canonical/lxd/lxd/instance/drivers/driver_qemu.go#L6436)
+
+This function controls the sending of a migration, checking for stateful support, waiting for connections, performing checks, and sending a migration offer. When performing a intra-cluster same-name migration, steps are taken to prevent corruption of volatile device configuration keys during start and stop of the instance on both source and target.
```

Comment:
```suggestion
This function controls the sending of a migration, checking for stateful support, waiting for connections, performing checks, and sending a migration offer. When performing an intra-cluster same-name migration, steps are taken to prevent corruption of volatile device configuration keys during the start and stop of the instance on both source and target.
```